From 8adef53e942743a0305625d4748d204d80d263b9 Mon Sep 17 00:00:00 2001
From: Felix Radensky <felix.r@variscite.com>
Date: Mon, 11 May 2020 09:05:19 +0300
Subject: [PATCH 02/31] imx8m: drivers/ddr: Change padding of DDR4 and LPDDR4
 DMEM firmware

DMEM firmware files are padded to 16KB, wasting precious space
in processor internal RAM (TMU). The actual size of these files
is less than 2KB, so padding them to 4KB is sufficient.

This patch changes the expected size of DMEM firmware.
It relies on boot image creation tools to generate 4KB padding.
---
 drivers/ddr/imx/imx8m/helper.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/ddr/imx/imx8m/helper.c b/drivers/ddr/imx/imx8m/helper.c
index a6f1814131..e2228927b9 100644
--- a/drivers/ddr/imx/imx8m/helper.c
+++ b/drivers/ddr/imx/imx8m/helper.c
@@ -16,8 +16,8 @@
 DECLARE_GLOBAL_DATA_PTR;
 
 #define IMEM_LEN 32768 /* byte */
-#define DMEM_LEN 16384 /* byte */
-#define IMEM_2D_OFFSET	49152
+#define DMEM_LEN 4096  /* byte */
+#define IMEM_2D_OFFSET	36864
 
 #define IMEM_OFFSET_ADDR 0x00050000
 #define DMEM_OFFSET_ADDR 0x00054000
-- 
2.25.1

